internal/runbits/runtimemsg.go

```diff
@@ -3,61 +3,92 @@ package runbits
 // Progress bar design
 //
 import (
+	"context"
 	"fmt"
+	"strconv"
+	"time"
 
-	"github.com/go-openapi/strfmt"
+	"github.com/thoas/go-funk"
+	"github.com/vbauerster/mpb/v6"
 
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
 	"github.com/ActiveState/cli/internal/output"
 	"github.com/ActiveState/cli/pkg/platform/runtime/artifact"
-	"github.com/ActiveState/cli/pkg/platform/runtime/setup"
+	"github.com/ActiveState/cli/pkg/platform/runtime/setup/events"
 )
 
 type RuntimeMessageHandler struct {
 	out output.Outputer
 }
 
 func NewRuntimeMessageHandler(out output.Outputer) *RuntimeMessageHandler {
-	return &RuntimeMessageHandler{out: out}
+	return &RuntimeMessageHandler{out}
 }
 
-func (r RuntimeMessageHandler) BuildStarting(total int) {
-	r.out.Notice(fmt.Sprintf("Build Starting: %d", total))
+func (rmh *RuntimeMessageHandler) ChangeSummary(artifacts artifact.ArtifactRecipeMap, requested artifact.ArtifactChangeset, changed artifact.ArtifactChangeset) error {
+	// currently we only print a change summary if we are adding exactly ONE package
+	if len(requested.Added) != 1 {
+		return nil
+	}
+
+	ar, ok := artifacts[requested.Added[0]]
+	if !ok {
+		return errs.New("Did not find requested artifact in ArtifactRecipeMap")
+	}
+
+	// the added (direct dependencies) of this artifact that are actually new in this project
+	addedDependencies := funk.Join(ar.Dependencies, changed.Added, funk.InnerJoin).([]artifact.ArtifactID)
+
+	rmh.out.Notice("")
+	rmh.out.Notice(locale.Tl(
+		"changesummary_title",
+		"[NOTICE]{{.V0}}[/RESET] includes {{.V1}} dependencies, for a combined total of {{.V2}} new dependencies.",
+		ar.Name, strconv.Itoa(len(addedDependencies)), strconv.Itoa(len(changed.Added)),
+	))
+	for i, dep := range addedDependencies {
+		depMapping, ok := artifacts[dep]
+		if !ok {
+			logging.Error("Could not find dependency %s in artifactsMap", dep)
+			continue
+		}
+		var depCount string
+		recDeps := artifact.RecursiveDependenciesFor(dep, artifacts)
```

Comment:
This should probably be passed to `ChangeSummary` as an argument instead.

Reply:
It would be quite expensive to compute this for every artifact.  That's why I preferred the on-demand version.